### ISSUE TITLE
Installing PyTorch/XLA 1.13 wheel failed with "version `GLIBC_2.28' not found".

### ISSUE THREAD

**Issue by vanbasten23** [pytorch/xla#4150]
## 🐛 Bug

Installing PyTorch/XLA 1.13 wheel with PyTorch 1.13 failed with "version `GLIBC_2.28' not found" on Ubuntu 18.04.06.

## To Reproduce
```
(base) jenkins@efd99baac2e8:/workspace$ python3
Python 3.7.13 (default, Mar 29 2022, 02:18:16)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import torch_xla.core.xla_model as xm
>>>
(base) jenkins@efd99baac2e8:/workspace$ yes | sudo pip3 uninstall torch_xla
The directory '/home/jenkins/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Cannot uninstall requirement torch-xla, not installed
(base) jenkins@efd99baac2e8:/workspace$ yes | pip3 uninstall torch_xla
Found existing installation: torch-xla 1.13
Uninstalling torch-xla-1.13:
  Would remove:
    /opt/conda/lib/python3.7/site-packages/torch_xla-1.13-py3.7-linux-x86_64.egg
Proceed (Y/n)?   Successfully uninstalled torch-xla-1.13
(base) jenkins@efd99baac2e8:/workspace$ yes | pip3 uninstall torch_xla
WARNING: Skipping torch-xla as it is not installed.
(base) jenkins@efd99baac2e8:/workspace$ yes | pip3 uninstall torch
Found existing installation: torch 1.13.0a0+gitaab841e
Uninstalling torch-1.13.0a0+gitaab841e:
  Would remove:
    /opt/conda/bin/convert-caffe2-to-onnx
    /opt/conda/bin/convert-onnx-to-caffe2
    /opt/conda/bin/torchrun
    /opt/conda/lib/python3.7/site-packages/caffe2
    /opt/conda/lib/python3.7/site-packages/functorch
    /opt/conda/lib/python3.7/site-packages/torch
    /opt/conda/lib/python3.7/site-packages/torch-1.13.0a0+gitaab841e-py3.7.egg-info
    /opt/conda/lib/python3.7/site-packages/torchgen
Proceed (Y/n)?   Successfully uninstalled torch-1.13.0a0+gitaab841e
(base) jenkins@efd99baac2e8:/workspace$ yes | pip3 uninstall torch
Found existing installation: torch 1.13.0a0+git8669f6d
Uninstalling torch-1.13.0a0+git8669f6d:
  Would remove:
    /opt/conda/lib/python3.7/site-packages/torch-1.13.0a0+git8669f6d-py3.7.egg-info
Proceed (Y/n)?   Successfully uninstalled torch-1.13.0a0+git8669f6d
(base) jenkins@efd99baac2e8:/workspace$ yes | pip3 uninstall torch
WARNING: Skipping torch as it is not installed.
(base) jenkins@efd99baac2e8:/workspace$ yes | pip3 uninstall torch_vision
WARNING: Skipping torch-vision as it is not installed.
(base) jenkins@efd99baac2e8:/workspace$ pip3 install torch_visionpip3 install --pre torch torchvision torchaudio -f https://download.pytorch.org/whl/test/cpu/torch_test.html^C
(base) jenkins@efd99baac2e8:/workspace$ pip3 install --pre torch torchvision torchaudio -f https://download.pytorch.org/whl/test/cpu/torch_test.html
Looking in links: https://download.pytorch.org/whl/test/cpu/torch_test.html
Collecting torch
  Downloading https://download.pytorch.org/whl/test/cpu/torch-1.13.0%2Bcpu-cp37-cp37m-linux_x86_64.whl (198.6 MB)
     |████████████████████████████████| 198.6 MB 87 kB/s
Collecting torchvision
  Downloading https://download.pytorch.org/whl/test/cpu/torchvision-0.14.0%2Bcpu-cp37-cp37m-linux_x86_64.whl (16.8 MB)
     |████████████████████████████████| 16.8 MB 40.1 MB/s
Collecting torchaudio
  Downloading https://download.pytorch.org/whl/test/cpu/torchaudio-0.13.0%2Bcpu-cp37-cp37m-linux_x86_64.whl (4.0 MB)
     |████████████████████████████████| 4.0 MB 775 kB/s
Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from torch) (3.10.0.2)
Requirement already satisfied: numpy in /opt/conda/lib/python3.7/site-packages (from torchvision) (1.18.5)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in /opt/conda/lib/python3.7/site-packages (from torchvision) (9.2.0)
Requirement already satisfied: requests in /opt/conda/lib/python3.7/site-packages (from torchvision) (2.28.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision) (1.26.10)
Requirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision) (3.3)
Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision) (2022.6.15)
Installing collected packages: torch, torchvision, torchaudio
Successfully installed torch-1.13.0+cpu torchaudio-0.13.0+cpu torchvision-0.14.0+cpu
(base) jenkins@efd99baac2e8:/workspace$ pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp38-cp38-linux_x86_64.whl
ERROR: torch_xla-1.13-cp38-cp38-linux_x86_64.whl is not a supported wheel on this platform.
(base) jenkins@efd99baac2e8:/workspace$ python -V
Python 3.7.13
(base) jenkins@efd99baac2e8:/workspace$ pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp37-cp37m-linux_x86_64.whl
Collecting torch-xla==1.13
  Downloading https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp37-cp37m-linux_x86_64.whl (154.0 MB)
     |████████████████████████████████| 154.0 MB 80.5 MB/s
Requirement already satisfied: absl-py>=1.0.0 in /opt/conda/lib/python3.7/site-packages (from torch-xla==1.13) (1.1.0)
Requirement already satisfied: cloud-tpu-client>=0.10.0 in /opt/conda/lib/python3.7/site-packages (from torch-xla==1.13) (0.10)
Requirement already satisfied: oauth2client in /opt/conda/lib/python3.7/site-packages (from cloud-tpu-client>=0.10.0->torch-xla==1.13) (4.1.3)
Requirement already satisfied: google-api-python-client==1.8.0 in /opt/conda/lib/python3.7/site-packages (from cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.8.0)
Requirement already satisfied: google-api-core<2dev,>=1.13.0 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.31.6)
Requirement already satisfied: six<2dev,>=1.6.1 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.16.0)
Requirement already satisfied: google-auth-httplib2>=0.0.3 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.1.0)
Requirement already satisfied: uritemplate<4dev,>=3.0.0 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.0.1)
Requirement already satisfied: httplib2<1dev,>=0.9.2 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.20.4)
Requirement already satisfied: google-auth>=1.4.1 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.35.0)
Requirement already satisfied: protobuf<4.0.0dev,>=3.12.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.20.1)
Requirement already satisfied: packaging>=14.3 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (21.3)
Requirement already satisfied: pytz in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2022.1)
Requirement already satisfied: requests<3.0.0dev,>=2.18.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2.28.0)
Requirement already satisfied: googleapis-common-protos<2.0dev,>=1.6.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.56.3)
Requirement already satisfied: setuptools>=40.3.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (61.2.0)
Requirement already satisfied: cachetools<5.0,>=2.0.0 in /opt/conda/lib/python3.7/site-packages (from google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (4.2.4)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /opt/conda/lib/python3.7/site-packages (from google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.2.8)
Requirement already satisfied: rsa<5,>=3.1.4 in /opt/conda/lib/python3.7/site-packages (from google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (4.8)
Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /opt/conda/lib/python3.7/site-packages (from httplib2<1dev,>=0.9.2->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.0.9)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /opt/conda/lib/python3.7/site-packages (from pyasn1-modules>=0.2.1->google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.4.8)
Requirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.3)
Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2022.6.15)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.26.10)
Installing collected packages: torch-xla
Successfully installed torch-xla-1.13

(base) jenkins@efd99baac2e8:/workspace$ pip freeze | grep -e libtpu -e torch
libtpu-nightly @ https://storage.googleapis.com/cloud-tpu-tpuvm-artifacts/wheels/libtpu-nightly/libtpu_nightly-0.1.dev20220816-py3-none-any.whl
torch==1.13.0+cpu
torch-xla @ https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp37-cp37m-linux_x86_64.whl
torchaudio==0.13.0+cpu
torchvision==0.14.0+cpu

(base) jenkins@efd99baac2e8:/workspace$ pip freeze | grep -e libtpu -e torch
libtpu-nightly @ https://storage.googleapis.com/cloud-tpu-tpuvm-artifacts/wheels/libtpu-nightly/libtpu_nightly-0.1.dev20220816-py3-none-any.whl
torch==1.13.0+cpu
torch-xla @ https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp37-cp37m-linux_x86_64.whl
torchaudio==0.13.0+cpu
torchvision==0.14.0+cpu

(base) jenkins@efd99baac2e8:/workspace$ python3 xla/test/test_train_mp_imagenet.py --fake_data --model=resnet50 --num_epochs=1
Traceback (most recent call last):
  File "xla/test/test_train_mp_imagenet.py", line 44, in <module>
    import schedulers
  File "/workspace/xla/test/schedulers.py", line 2, in <module>
    import torch_xla.test.test_utils as test_utils
  File "/opt/conda/lib/python3.7/site-packages/torch_xla/__init__.py", line 114, in <module>
    import _XLAC
ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /opt/conda/lib/python3.7/site-packages/torch_xla/lib/libxla_computation_client.so)

(base) jenkins@efd99baac2e8:/workspace$ strings /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 | grep GLIBC
strings: '/usr/lib/arm-linux-gnueabihf/libstdc++.so.6': No such file
```

## Expected behavior

Installing the 1.13 wheel shouldn't fail.

## Environment
```
 (base) jenkins@efd99baac2e8:/workspace$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.6 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

**Comment by vanbasten23** [comment]
This is caused by a PR from meta. From Wei Wang from Meta:
"this may be related to pytorch/xla depending on nodejs lts version. Coincidentally, nodejs lts version gets updated from v16.18.0 to v18.12.0, causing glibc issues on older platforms. https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#toolchain-and-compiler-upgrades Currently we are attempting to fix this by locking nodejs to an older version like we used to be using the "lts" version."

**Comment by vanbasten23** [comment]
Wei has submitted a PR https://github.com/pytorch/xla/pull/4125 .

**Comment by JackCaoG** [comment]
what's the next step here?

**Comment by vanbasten23** [comment]
I have validated on my cloudtop (same environment as in comment#1) and the error is gone. It failed with a different error now:
```
OSError: /opt/conda/lib/python3.7/site-packages/torch/lib/../../nvidia/cublas/lib/libcublas.so.11: symbol cublasLtGetStatusString version libcublasLt.so.11 not defined in file libcublasLt.so.11 with link time reference
```

Steps to repro:
```
(base) jenkins@efd99baac2e8:/workspace/pytorch$ yes | pip3 uninstall torch_xla
Found existing installation: torch-xla 1.13
Uninstalling torch-xla-1.13:
  Would remove:
    /opt/conda/lib/python3.7/site-packages/_XLAC.cpython-37m-x86_64-linux-gnu.so
    /opt/conda/lib/python3.7/site-packages/test/*
    /opt/conda/lib/python3.7/site-packages/torch_xla-1.13.dist-info/*
    /opt/conda/lib/python3.7/site-packages/torch_xla/*
Proceed (Y/n)?   Successfully uninstalled torch-xla-1.13
(base) jenkins@efd99baac2e8:/workspace/pytorch$ yes | pip3 uninstall torch_xla
WARNING: Skipping torch-xla as it is not installed.
(base) jenkins@efd99baac2e8:/workspace/pytorch$ yes | pip3 uninstall torch
Found existing installation: torch 1.13.0+cpu
Uninstalling torch-1.13.0+cpu:
  Would remove:
    /opt/conda/bin/convert-caffe2-to-onnx
    /opt/conda/bin/convert-onnx-to-caffe2
    /opt/conda/bin/torchrun
    /opt/conda/lib/python3.7/site-packages/caffe2/*
    /opt/conda/lib/python3.7/site-packages/functorch/*
    /opt/conda/lib/python3.7/site-packages/torch-1.13.0+cpu.dist-info/*
    /opt/conda/lib/python3.7/site-packages/torch/*
    /opt/conda/lib/python3.7/site-packages/torchgen/*
Proceed (Y/n)?   Successfully uninstalled torch-1.13.0+cpu
(base) jenkins@efd99baac2e8:/workspace/pytorch$ yes | pip3 uninstall torch
WARNING: Skipping torch as it is not installed.
(base) jenkins@efd99baac2e8:/workspace/pytorch$ yes | pip3 uninstall torch_vision
WARNING: Skipping torch-vision as it is not installed.
(base) jenkins@efd99baac2e8:/workspace/pytorch$ pip3 install torch==1.13
Collecting torch==1.13
  Downloading torch-1.13.0-cp37-cp37m-manylinux1_x86_64.whl (890.2 MB)
     |████████████████████████████████| 890.2 MB 5.1 kB/s
Collecting nvidia-cublas-cu11==11.10.3.66
  Downloading nvidia_cublas_cu11-11.10.3.66-py3-none-manylinux1_x86_64.whl (317.1 MB)
     |████████████████████████████████| 317.1 MB 39 kB/s
Collecting nvidia-cuda-nvrtc-cu11==11.7.99
  Downloading nvidia_cuda_nvrtc_cu11-11.7.99-2-py3-none-manylinux1_x86_64.whl (21.0 MB)
     |████████████████████████████████| 21.0 MB 63.4 MB/s
Collecting nvidia-cuda-runtime-cu11==11.7.99
  Downloading nvidia_cuda_runtime_cu11-11.7.99-py3-none-manylinux1_x86_64.whl (849 kB)
     |████████████████████████████████| 849 kB 50.6 MB/s
Collecting nvidia-cudnn-cu11==8.5.0.96
  Downloading nvidia_cudnn_cu11-8.5.0.96-2-py3-none-manylinux1_x86_64.whl (557.1 MB)
     |████████████████████████████████| 557.1 MB 17 kB/s
Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from torch==1.13) (3.10.0.2)
Requirement already satisfied: setuptools in /opt/conda/lib/python3.7/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch==1.13) (61.2.0)
Requirement already satisfied: wheel in /opt/conda/lib/python3.7/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch==1.13) (0.37.1)
Installing collected packages: nvidia-cublas-cu11, nvidia-cudnn-cu11, nvidia-cuda-runtime-cu11, nvidia-cuda-nvrtc-cu11, torch
Successfully installed nvidia-cublas-cu11-11.10.3.66 nvidia-cuda-nvrtc-cu11-11.7.99 nvidia-cuda-runtime-cu11-11.7.99 nvidia-cudnn-cu11-8.5.0.96 torch-1.13.0
(base) jenkins@efd99baac2e8:/workspace/pytorch$ pip3 install torchvision==0.14.0
Requirement already satisfied: torchvision==0.14.0 in /opt/conda/lib/python3.7/site-packages (0.14.0+cpu)
Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from torchvision==0.14.0) (3.10.0.2)
Requirement already satisfied: numpy in /opt/conda/lib/python3.7/site-packages (from torchvision==0.14.0) (1.18.5)
Requirement already satisfied: requests in /opt/conda/lib/python3.7/site-packages (from torchvision==0.14.0) (2.28.0)
Requirement already satisfied: torch==1.13.0 in /opt/conda/lib/python3.7/site-packages (from torchvision==0.14.0) (1.13.0)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in /opt/conda/lib/python3.7/site-packages (from torchvision==0.14.0) (9.2.0)
Requirement already satisfied: nvidia-cublas-cu11==11.10.3.66 in /opt/conda/lib/python3.7/site-packages (from torch==1.13.0->torchvision==0.14.0) (11.10.3.66)
Requirement already satisfied: nvidia-cuda-runtime-cu11==11.7.99 in /opt/conda/lib/python3.7/site-packages (from torch==1.13.0->torchvision==0.14.0) (11.7.99)
Requirement already satisfied: nvidia-cuda-nvrtc-cu11==11.7.99 in /opt/conda/lib/python3.7/site-packages (from torch==1.13.0->torchvision==0.14.0) (11.7.99)
Requirement already satisfied: nvidia-cudnn-cu11==8.5.0.96 in /opt/conda/lib/python3.7/site-packages (from torch==1.13.0->torchvision==0.14.0) (8.5.0.96)
Requirement already satisfied: wheel in /opt/conda/lib/python3.7/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch==1.13.0->torchvision==0.14.0) (0.37.1)
Requirement already satisfied: setuptools in /opt/conda/lib/python3.7/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch==1.13.0->torchvision==0.14.0) (61.2.0)
Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision==0.14.0) (2022.6.15)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision==0.14.0) (1.26.10)
Requirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision==0.14.0) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests->torchvision==0.14.0) (3.3)
(base) jenkins@efd99baac2e8:/workspace/pytorch$ pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp37-cp37m-linux_x86_64.whl
Collecting torch-xla==1.13
  Downloading https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp37-cp37m-linux_x86_64.whl (154.7 MB)
     |████████████████████████████████| 154.7 MB 94.3 MB/s
Requirement already satisfied: cloud-tpu-client>=0.10.0 in /opt/conda/lib/python3.7/site-packages (from torch-xla==1.13) (0.10)
Requirement already satisfied: absl-py>=1.0.0 in /opt/conda/lib/python3.7/site-packages (from torch-xla==1.13) (1.1.0)
Requirement already satisfied: oauth2client in /opt/conda/lib/python3.7/site-packages (from cloud-tpu-client>=0.10.0->torch-xla==1.13) (4.1.3)
Requirement already satisfied: google-api-python-client==1.8.0 in /opt/conda/lib/python3.7/site-packages (from cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.8.0)
Requirement already satisfied: httplib2<1dev,>=0.9.2 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.20.4)
Requirement already satisfied: google-auth>=1.4.1 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.35.0)
Requirement already satisfied: uritemplate<4dev,>=3.0.0 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.0.1)
Requirement already satisfied: google-auth-httplib2>=0.0.3 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.1.0)
Requirement already satisfied: six<2dev,>=1.6.1 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.16.0)
Requirement already satisfied: google-api-core<2dev,>=1.13.0 in /opt/conda/lib/python3.7/site-packages (from google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.31.6)
Requirement already satisfied: pytz in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2022.1)
Requirement already satisfied: requests<3.0.0dev,>=2.18.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2.28.0)
Requirement already satisfied: setuptools>=40.3.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (61.2.0)
Requirement already satisfied: googleapis-common-protos<2.0dev,>=1.6.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.56.3)
Requirement already satisfied: protobuf<4.0.0dev,>=3.12.0 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.20.1)
Requirement already satisfied: packaging>=14.3 in /opt/conda/lib/python3.7/site-packages (from google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (21.3)
Requirement already satisfied: rsa<5,>=3.1.4 in /opt/conda/lib/python3.7/site-packages (from google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (4.8)
Requirement already satisfied: cachetools<5.0,>=2.0.0 in /opt/conda/lib/python3.7/site-packages (from google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (4.2.4)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /opt/conda/lib/python3.7/site-packages (from google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.2.8)
Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /opt/conda/lib/python3.7/site-packages (from httplib2<1dev,>=0.9.2->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.0.9)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /opt/conda/lib/python3.7/site-packages (from pyasn1-modules>=0.2.1->google-auth>=1.4.1->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (0.4.8)
Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2022.6.15)
Requirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (3.3)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core<2dev,>=1.13.0->google-api-python-client==1.8.0->cloud-tpu-client>=0.10.0->torch-xla==1.13) (1.26.10)
Installing collected packages: torch-xla
Successfully installed torch-xla-1.13
(base) jenkins@efd99baac2e8:/workspace/pytorch$ cd ..
(base) jenkins@efd99baac2e8:/workspace$ ls
'=3.13'   install_conda.sh   pytorch   xla
(base) jenkins@efd99baac2e8:/workspace$ python3 xla/test/test_train_mp_imagenet.py --fake_data --model=resnet50 --num_epochs=1
Traceback (most recent call last):
  File "xla/test/test_train_mp_imagenet.py", line 44, in <module>
    import schedulers
  File "/workspace/xla/test/schedulers.py", line 2, in <module>
    import torch_xla.test.test_utils as test_utils
  File "/opt/conda/lib/python3.7/site-packages/torch_xla/__init__.py", line 105, in <module>
    import torch
  File "/opt/conda/lib/python3.7/site-packages/torch/__init__.py", line 191, in <module>
    _load_global_deps()
  File "/opt/conda/lib/python3.7/site-packages/torch/__init__.py", line 153, in _load_global_deps
    ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
  File "/opt/conda/lib/python3.7/ctypes/__init__.py", line 364, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /opt/conda/lib/python3.7/site-packages/torch/lib/../../nvidia/cublas/lib/libcublas.so.11: symbol cublasLtGetStatusString version libcublasLt.so.11 not defined in file libcublasLt.so.11 with link time reference
```

**Comment by JackCaoG** [comment]
OK I am a bit confuse, seems like error is caused by the `node.js v18` which the fix is merged to master and 1.13. Do you know why do we still see this error on release wheel?

**Comment by vanbasten23** [comment]
Per our discussion, on both 1vm and 2vm, there is a GLIBC_2.28 installed so we should be good there.

Our build machine is on debian 10, which seems like it come with glibc 2.28. When using docker images, we should use the corresponding docker image listed in https://github.com/pytorch/xla#-available-images-and-wheels. IOW, don't use [gcr.io/tpu-pytorch/xla_base:latest](http://gcr.io/tpu-pytorch/xla_base:latest%60) which doesn't have glibc_2.28.

Conclusion is pt/xla 1.13 wheel requires 2.28 might not be an issue, we are not many linux compatible so as long as our wheel works on the platform we need that's fine. If this issue comes up with colab we can revisit.
